controllers/openstackcluster_controller.go

```diff
@@ -239,20 +241,28 @@ func deleteBastion(scope scope.Scope, cluster *clusterv1.Cluster, openStackClust
 
 		instanceSpec := bastionToInstanceSpec(openStackCluster, cluster.Name)
 
-		if err = computeService.DeleteInstance(openStackCluster, openStackCluster, instanceStatus, instanceSpec); err != nil {
+		if err = computeService.DeleteInstance(openStackCluster, instanceStatus, instanceSpec); err != nil {
 			handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to delete bastion: %w", err))
 			return fmt.Errorf("failed to delete bastion: %w", err)
 		}
 	}
 
-	openStackCluster.Status.Bastion = nil
-
-	if err = networkingService.DeleteBastionSecurityGroup(openStackCluster, fmt.Sprintf("%s-%s", cluster.Namespace, cluster.Name)); err != nil {
-		handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to delete bastion security group: %w", err))
-		return fmt.Errorf("failed to delete bastion security group: %w", err)
+	if openStackCluster.Status.Bastion != nil && openStackCluster.Status.Bastion.PortsStatus != nil {
+		trunkEnabled, err := networkingService.IsTrunkExtSupported()
+		if err != nil {
+			return err
+		}
+		scope.Logger().Info("Deleting ports for bastion host")
+		for _, port := range openStackCluster.Status.Bastion.PortsStatus {
+			if err := networkingService.DeleteInstanceTrunkAndPort(openStackCluster, port, trunkEnabled); err != nil {
+				handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to delete port: %w", err))
+				return fmt.Errorf("failed to delete port: %w", err)
+			}
+		}
+		openStackCluster.Status.Bastion.PortsStatus = nil
```

Comment:
Matt: I know you [preferred](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1708/files#diff-f139d325a459bd4476f19197576c8b5b106d231872e994790c4e0842e538f264R45-R49) to reimplement the bastion to create an OpenStackMachine object and rely on the machine controller instead.
I disagree on this one, re-implementing the bastion is an effort on its own, that I don't want to make in this PR.

I think it's ok now to have Ports status in OpenStack.Cluster.Status.Bastion.

Reply:
We chatted offline and he agreed that for now we'll proceed Bastion status.

Reply:
I've done a 180 on that: I no longer think we can use an OpenStackMachine to create the Bastion because of the potential to interact poorly with other tooling. A Machine which never gets a NodeRef would be an anomaly.

The way we're tackling this is by putting specific parts of OpenStackMachine status in the Bastion's status.